### PR TITLE
feat: Implement MinIO Tenant CRD health check

### PR DESCRIPTION
**Pull request by dnskr**
The PR adds health check for MinIO Tenant CRD which is used by [MinIO Operator](https://github.com/minio/operator)
List of [Standard Status messages for Tenant](https://github.com/minio/operator/blob/b5ad899041245b6f8f49009a1263cf3747687897/pkg/controller/cluster/main-controller.go#L114-L142)

---
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by dnskr** [comment]
Hi @jannfis, @alexmt and @jessesuen! Could anyone please review the PR?

File: resource_customizations/minio.min.io/Tenant/health.lua [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            health_status.status = "Degraded"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        health_status.status = "Unknown"
+        health_status.message = obj.status.currentState
+        return health_status
+    end
+end
+health_status.status = "Missing"
```
Comment (alexmt) [inline resource_customizations/minio.min.io/Tenant/health.lua:44]: Health check should not return `Unknown` or `Missing`. These are internal statuses. `Unknown` means error and `Missing` means expected resource not found in the cluster. To be consistent with other checks the health check need to return `Progressing`

**Review by alexmt — CHANGES_REQUESTED** [review]
Thank you for contribution @dnskr ! Added one minor comment. Can you take a look please?

File: resource_customizations/minio.min.io/Tenant/health.lua (hunk 1 above)
Comment (dnskr) [reply resource_customizations/minio.min.io/Tenant/health.lua]: Fixed!
Could status descriptions be added to the [documentation](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/)?

**Review by alexmt — APPROVED** [review]
LGTM. Thank you!

File: resource_customizations/minio.min.io/Tenant/health.lua (hunk 1 above)
Comment (alexmt) [reply resource_customizations/minio.min.io/Tenant/health.lua]: Agree. Created PR with docs changes: https://github.com/argoproj/argo-cd/pull/6692

**Comment by dnskr** [comment]
Hi @jannfis and @alexmt!
I found this PR was merged but has not been included in the release 2.0.5
Am I right that `needs-verification` label prevented it? What should be done to release the health check?

**Comment by alexmt** [comment]
Hello @dnskr , 

Only bugfixes get cherry-picked into patch releases. New features will be available in 2.1 release - I'm hoping to create first release candidate today and GA should be available in ~week.

**Comment by dnskr** [comment]
Got it. Thank you for the quick answer!

**Pull request merged**